loguru_mypy/__init__.py

```diff
@@ -47,6 +50,35 @@ class Opts(t.NamedTuple):
 }  # type: te.Final
 
 
+def _check_str_format_call(log_msg_expr: StrExpr, ctx: MethodContext) -> None:
+    """ Taps into mypy to typecheck something like this:
+
+        ```py
+        logger.debug('The bar is "{my_foo.bar}"', my_foo=foo)
+        ```
+
+        as if it was written like this:
+
+        ```py
+        logger.debug('The bar is "{my_foo.bar}"'.format(my_foo=foo))
+        ```
+    """
+    call_expr = CallExpr(
+        callee=MemberExpr(expr=log_msg_expr, name='format'),
+        args=ctx.args[1] + ctx.args[2],
+        arg_kinds=ctx.arg_kinds[1] + ctx.arg_kinds[2],
+        arg_names=ctx.arg_names[1] + ctx.arg_names[2],
+    )
+    call_expr.set_line(log_msg_expr)
+
+    # WARNING: `ctx.api` *is* technically a `mypy.checker.TypeChecker` so the cast is
+    # safe to make, however, mypy says this should be an implementation detail.
+    # So, anything that's not part of the `CheckerPluginInterface` should be expected to
+    # change. See https://github.com/python/mypy/issues/6617
+    type_checker = t.cast(TypeChecker, ctx.api)
+    type_checker.expr_checker.visit_call_expr(call_expr)
```

Comment:
PS. Can add a `try...except` here to catch `AttributeError`? An error we log or re-throw (up to you) should basically say that someone might be using incompatible mypy version.

Reply:
Good idea, but an exception would bubble up at parse time and completely crash mypy... On the other hand, a mypy error message might look weird to the user (as it's not really a problem with their code) but would let mypy continue on with the rest of the analysis.

Also, how can we test that code path? I don't know of a version of mypy that would trigger that

I'll go with the mypy message message for now, tell me what you think.

Reply:
Version of mypy is related but external topic. I've already created an issue for that which also I am happy to accept PR.
I am afraid that with ongoing craziness (also work) I might just lose it out of focus :(. Anyway...it seems like entire world gos crazy nowadays, so yeah :D